src/libcore/iter/traits.rs

```diff
@@ -970,9 +970,11 @@ impl<'a, I: FusedIterator + ?Sized> FusedIterator for &'a mut I {}
 /// The iterator reports a size hint where it is either exact
 /// (lower bound is equal to upper bound), or the upper bound is [`None`].
 /// The upper bound must only be [`None`] if the actual iterator length is
-/// larger than [`usize::MAX`].
+/// larger than [`usize::MAX`]. In that case, the lower bound must be
+/// [`usize::MAX`], resulting in a [`.size_hint`] of `(usize::MAX, None)`.
```

Comment:
:+1: to this.  Probably worth mentioning in release notes just in case, though?

Reply:
How would I go about doing so? I don't know if it's really needed because TrustedLen is unstable anyways. But better safe than sorry.

Reply:
I think for now it just means someone with tagging permissions (thus not me, at least) putting `relnotes` on this.

Reply:
It doesn't have an impact to users of stable Rust, so then it's normally not in any release notes.

Reply:
Ah, makes sense.  I guess nightly finds out by watching the TWiR PRs section.